### ISSUE TITLE
False positive for scientific notation

### ISSUE THREAD

**Issue by nyuszika7h** [adrienverge/yamllint#495]
Given the following YAML file:
```yaml
foo: 1e2a
```

Running `yamllint -d 'rules: {float-values: {forbid-scientific-notation: true}}' test.yaml` produces this error:
```
test.yaml
  1:6       error    forbidden scientific notation "1e2a"  (float-values)
```

This is incorrect, as it's actually parsed as a string:
```
❯ python -c 'import yaml; print(yaml.safe_load(open("test.yaml")))'
{'foo': '1e2a'}
❯ python -c 'from ruamel.yaml import YAML; print(YAML().load(open("test.yaml")))'
ordereddict([('foo', '1e2a')])
```

---

```
❯ yamllint --version
yamllint 1.28.0     
```

**Comment by adrienverge** [comment]
Hello @nyuszika7h, thanks for the report!

I can confirm the bug, and it actually occurs on all options of the `float-values` rule:
```yaml
- .1two3
- 1e2a
- .NaNa
- .Infinit∞
```

```
yamllint -d 'rules: {float-values: {require-numeral-before-decimal: true,
                                    forbid-scientific-notation: true,
                                    forbid-nan: true,
                                    forbid-inf: true}}' /tmp/file.yaml

/tmp/file.yaml
  1:3       error    forbidden decimal missing 0 prefix ".1two3"  (float-values)
  2:3       error    forbidden scientific notation "1e2a"  (float-values)
  3:3       error    forbidden not a number value ".NaNa"  (float-values)
  4:3       error    forbidden infinite value ".Infinit∞"  (float-values)
```

I guess the writing of tests wasn't thorough enough on https://github.com/adrienverge/yamllint/pull/465.

I'll work on a fix.
